addon/lib/Feature.jsm

```diff
@@ -214,6 +242,52 @@ class Feature {
     });
   }
 
+  async initBehaviorSummary() {
+    const initialized = (await Storage.has("behavior-summary"));
+    if (initialized) return;
+
+    // does the user have any of the top adblockers?
+    const ADBLOCKER_ID_LIST = [
+                                "{d10d0bf8-f5b5-c8b4-a8b2-2b9879e08c5d}", // AdblockPlus
+                                "uBlock0@raymondhill.net", // uBlock Origin
+                                "jid1-NIfFY2CA8fy1tg@jetpack" // Adblock for Firefox
+                              ];
+    let containsAddon = false;
+    AddonManager.getAllAddons().then((addons) => {
+      for (let addon of addons) {
+        if (ADBLOCKER_ID_LIST.includes(addon.id)) containsAddon = true;
+      }
+    })
+
+    // co-variates
+    const oldestTimestamp = await ((new ProfileAge()).getOldestProfileTimestamp())
+    const ProfileAgeDays = Math.round((Date.now() - oldestTimestamp) / (1000 * 3600 * 24));
+    const dntEnabled = Services.prefs.getBoolPref("privacy.donottrackheader.enabled");
+    const historyEnabled = Services.prefs.getBoolPref("places.history.enabled");
+    const appUpdateEnabled = Services.prefs.getBoolPref("app.update.enabled");
+
+    await Storage.create("behavior-summary", {
+      reject: "false",
+      intro_accept: "false",
+      intro_reject: "false",
+      badge_clicks: String(0),
+      panel_open_times: String([]),
+      panel_open_times_median: String(0),
+      panel_open_times_mean: String(0),
+      new_tab_open_times: String([]),
+      new_tab_open_times_median: String(0),
+      new_tab_open_times_mean: String(0),
+      page_action_counter: String([]),
```

Comment:
Sorry, I'm not too knowledgeable on Storage vs `String()` constructors, but...
IIUC, calling `String([])` will return an empty string, `""`. Whereas `JSON.stringify([])` will return `"[]"`. I have no idea if any of that matters, I guess it all depends on who consumes the data at the other end (or how we retrieve and rehydrate it from storage):

```js
String([]); // ""
String(["one fish", "two fish"]) //"one fish,two fish"

JSON.stringify([]) // "[]"
JSON.stringify(["red fish", "blue fish"]) // "[\"red fish\",\"blue fish\"]"
```

Reply:
I agree that JSON.stringify is a better choice.

Reply:
I don't know about better or worse (or write our own stringifier, if we think we'd use it in multiple places). But I do know calling `String(true)`, `String([])`, and `String(0)` gives different results than `JSON.stringify(true)`, `JSON.stringify([])`, and `JSON.stringify(0)` so we'd need to make sure our Telemetry data (or whatever we're sending this packet to) is still valid at the other end.